gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -0,0 +1,155 @@
+                    task.getGradleJdksSetupJar()
+                            .fileProvider(generateGradleJdkConfigs
+                                    .map(GradleJdkConfigsTask::getOutputGradleDirectory)
+                                    .flatMap(dirProp -> dirProp.getAsFile().map(file -> file.toPath()
+                                            .resolve("gradle-jdks-setup.jar")
+                                            .toFile())));
+                    task.getPatchedGradlewScript()
+                            .set(rootProject.file(
+                                    rootProject.getRootDir().toPath().resolve("gradlew")));
+                    task.getPatchedGradleWrapperJar()
+                            .set(rootProject.file(
+                                    rootProject.getRootDir().toPath().resolve("gradle/wrapper/gradle-wrapper.jar")));
+                    task.getGenerate().set(true);
+                });
+        TaskProvider<GradleWrapperPatcherTask> checkWrapperPatcher = rootProject
+                .getTasks()
+                .register("checkWrapperPatcher", GradleWrapperPatcherTask.class, task -> {
+                    task.getOriginalGradlewScript().fileProvider(wrapperPatcherTask.map(wrapperPatcher -> wrapperPatcher
+                            .getPatchedGradlewScript()
+                            .get()
+                            .getAsFile()));
+                    task.getOriginalGradleWrapperJar()
+                            .fileProvider(wrapperPatcherTask.map(wrapperPatcher -> wrapperPatcher
+                                    .getPatchedGradleWrapperJar()
+                                    .getAsFile()
+                                    .get()));
+                    task.getBuildDir().set(task.getTemporaryDir());
+                    task.getGradleJdksSetupJar()
+                            .fileProvider(generateGradleJdkConfigs
+                                    .map(GradleJdkConfigsTask::getOutputGradleDirectory)
+                                    .flatMap(dirProp -> dirProp.getAsFile().map(file -> file.toPath()
+                                            .resolve("gradle-jdks-setup.jar")
+                                            .toFile())));
+                    task.getPatchedGradlewScript()
+                            .set(rootProject.getLayout().getBuildDirectory().file("checkWrapperPatcher/gradlew"));
+                    task.getPatchedGradleWrapperJar()
+                            .set(rootProject
+                                    .getLayout()
+                                    .getBuildDirectory()
+                                    .file("checkWrapperPatcher/gradle-wrapper.jar"));
+                    task.getGenerate().set(false);
+                });
+
+        rootProject
+                .getTasks()
+                .named(LifecycleBasePlugin.CHECK_TASK_NAME)
+                .configure(check -> check.dependsOn(checkGradleJdkConfigs, checkWrapperPatcher));
+
```

Comment:
Pretty sure we're going to want a single `setupJdks` task or something that depends on the two generate tasks so we can call just one task in the rollout excavator.